// create-payload-app/dist/lib/create-project.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    createProject: function() {
        return createProject;
    },
    updatePackageJSON: function() {
        return updatePackageJSON;
    }
});
const _chalk = /*#__PURE__*/ _interop_require_default(require("chalk"));
const _degit = /*#__PURE__*/ _interop_require_default(require("degit"));
const _execa = /*#__PURE__*/ _interop_require_default(require("execa"));
const _fsextra = /*#__PURE__*/ _interop_require_default(require("fs-extra"));
const _ora = /*#__PURE__*/ _interop_require_default(require("ora"));
const _path = /*#__PURE__*/ _interop_require_default(require("path"));
const _log = require("../utils/log");
const _configurepayloadconfig = require("./configure-payload-config");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
async function createOrFindProjectDir(projectDir) {
    const pathExists = await _fsextra.default.pathExists(projectDir);
    if (!pathExists) {
        await _fsextra.default.mkdir(projectDir);
    }
}
async function installDeps(args) {
    const { cliArgs, packageManager, projectDir } = args;
    if (cliArgs['--no-deps']) {
        return true;
    }
    let installCmd = 'npm install --legacy-peer-deps';
    if (packageManager === 'yarn') {
        installCmd = 'yarn';
    } else if (packageManager === 'pnpm') {
        installCmd = 'pnpm install';
    }
    try {
        await _execa.default.command(installCmd, {
            cwd: _path.default.resolve(projectDir)
        });
        return true;
    } catch (err) {
        console.log({
            err
        });
        return false;
    }
}
async function createProject(args) {
    const { cliArgs, dbDetails, packageManager, projectDir, projectName, template } = args;
    await createOrFindProjectDir(projectDir);
    console.log(`\n  Creating project in ${_chalk.default.green(_path.default.resolve(projectDir))}\n`);
    if ('url' in template) {
        const emitter = (0, _degit.default)(template.url);
        await emitter.clone(projectDir);
    }
    const spinner = (0, _ora.default)('Checking latest Payload version...').start();
    await updatePackageJSON({
        projectDir,
        projectName
    });
    await (0, _configurepayloadconfig.configurePayloadConfig)({
        dbDetails,
        projectDir
    });
    // Remove yarn.lock file. This is only desired in Payload Cloud.
    const lockPath = _path.default.resolve(projectDir, 'yarn.lock');
    if (_fsextra.default.existsSync(lockPath)) {
        await _fsextra.default.remove(lockPath);
    }
    spinner.text = 'Installing dependencies...';
    const result = await installDeps({
        cliArgs,
        packageManager,
        projectDir
    });
    spinner.stop();
    spinner.clear();
    if (result) {
        (0, _log.success)('Dependencies installed');
    } else {
        (0, _log.error)('Error installing dependencies');
    }
}
async function updatePackageJSON(args) {
    const { projectDir, projectName } = args;
    const packageJsonPath = _path.default.resolve(projectDir, 'package.json');
    try {
        const packageObj = await _fsextra.default.readJson(packageJsonPath);
        packageObj.name = projectName;
        await _fsextra.default.writeJson(packageJsonPath, packageObj, {
            spaces: 2
        });
    } catch (err) {
        (0, _log.warning)('Unable to update name in package.json');
    }
}
