// create-payload-app/dist/lib/packages.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    dbPackages: function() {
        return dbPackages;
    },
    bundlerPackages: function() {
        return bundlerPackages;
    },
    editorPackages: function() {
        return editorPackages;
    }
});
const mongodbReplacement = {
    importReplacement: "import { mongooseAdapter } from '@payloadcms/db-mongodb'",
    packageName: '@payloadcms/db-mongodb',
    // Replacement between `// database-adapter-config-start` and `// database-adapter-config-end`
    configReplacement: [
        '  db: mongooseAdapter({',
        '    url: process.env.DATABASE_URI,',
        '  }),'
    ],
    version: '^1.0.0'
};
const postgresReplacement = {
    configReplacement: [
        '  db: postgresAdapter({',
        '    pool: {',
        '      connectionString: process.env.DATABASE_URI,',
        '    },',
        '  }),'
    ],
    importReplacement: "import { postgresAdapter } from '@payloadcms/db-postgres'",
    packageName: '@payloadcms/db-postgres',
    version: '^0.x'
};
const dbPackages = {
    mongodb: mongodbReplacement,
    postgres: postgresReplacement
};
const webpackReplacement = {
    importReplacement: "import { webpackBundler } from '@payloadcms/bundler-webpack'",
    packageName: '@payloadcms/bundler-webpack',
    // Replacement of line containing `// bundler-config`
    configReplacement: '    bundler: webpackBundler(),',
    version: '^1.0.0'
};
const viteReplacement = {
    configReplacement: '  bundler: viteBundler(),',
    importReplacement: "import { viteBundler } from '@payloadcms/bundler-vite'",
    packageName: '@payloadcms/bundler-vite',
    version: '^0.x'
};
const bundlerPackages = {
    vite: viteReplacement,
    webpack: webpackReplacement
};
const editorPackages = {
    lexical: {
        configReplacement: '  editor: lexicalEditor({}),',
        importReplacement: "import { lexicalEditor } from '@payloadcms/richtext-lexical'",
        packageName: '@payloadcms/richtext-lexical',
        version: '^0.x'
    },
    slate: {
        configReplacement: '  editor: slateEditor({}),',
        importReplacement: "import { slateEditor } from '@payloadcms/richtext-slate'",
        packageName: '@payloadcms/richtext-slate',
        version: '^1.0.0'
    }
};
