// create-payload-app/dist/lib/parse-template.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "parseTemplate", {
    enumerable: true,
    get: function() {
        return parseTemplate;
    }
});
const _prompts = /*#__PURE__*/ _interop_require_default(require("prompts"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
async function parseTemplate(args, validTemplates) {
    if (args['--template']) {
        const templateName = args['--template'];
        const template = validTemplates.find((t)=>t.name === templateName);
        if (!template) throw new Error('Invalid template given');
        return template;
    }
    const response = await (0, _prompts.default)({
        name: 'value',
        choices: validTemplates.map((p)=>{
            return {
                description: p.description,
                title: p.name,
                value: p.name
            };
        }),
        message: 'Choose project template',
        type: 'select',
        validate: (value)=>!!value.length
    }, {
        onCancel: ()=>{
            process.exit(0);
        }
    });
    const template = validTemplates.find((t)=>t.name === response.value);
    if (!template) throw new Error('Template is undefined');
    return template;
}
