// create-payload-app/dist/lib/select-db.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "selectDb", {
    enumerable: true,
    get: function() {
        return selectDb;
    }
});
const _slugify = /*#__PURE__*/ _interop_require_default(require("@sindresorhus/slugify"));
const _prompts = /*#__PURE__*/ _interop_require_default(require("prompts"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const dbChoiceRecord = {
    mongodb: {
        dbConnectionPrefix: 'mongodb://127.0.0.1/',
        title: 'MongoDB',
        value: 'mongodb'
    },
    postgres: {
        dbConnectionPrefix: 'postgres://127.0.0.1:5432/',
        title: 'PostgreSQL (beta)',
        value: 'postgres'
    }
};
async function selectDb(args, projectName) {
    let dbType = undefined;
    if (args['--db']) {
        if (!Object.values(dbChoiceRecord).some((dbChoice)=>dbChoice.value === args['--db'])) {
            throw new Error(`Invalid database type given. Valid types are: ${Object.values(dbChoiceRecord).map((dbChoice)=>dbChoice.value).join(', ')}`);
        }
        dbType = args['--db'];
    } else {
        const dbTypeRes = await (0, _prompts.default)({
            name: 'value',
            choices: Object.values(dbChoiceRecord).map((dbChoice)=>{
                return {
                    title: dbChoice.title,
                    value: dbChoice.value
                };
            }),
            message: 'Select a database',
            type: 'select',
            validate: (value)=>!!value.length
        }, {
            onCancel: ()=>{
                process.exit(0);
            }
        });
        dbType = dbTypeRes.value;
    }
    const dbChoice = dbChoiceRecord[dbType];
    const dbUriRes = await (0, _prompts.default)({
        name: 'value',
        initial: `${dbChoice.dbConnectionPrefix}${projectName === '.' ? `payload-${getRandomDigitSuffix()}` : (0, _slugify.default)(projectName)}`,
        message: `Enter ${dbChoice.title.split(' ')[0]} connection string`,
        type: 'text',
        validate: (value)=>!!value.length
    }, {
        onCancel: ()=>{
            process.exit(0);
        }
    });
    return {
        dbUri: dbUriRes.value,
        type: dbChoice.value
    };
}
function getRandomDigitSuffix() {
    return (Math.random() * Math.pow(10, 6)).toFixed(0);
}
