// create-payload-app/dist/lib/templates.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    validateTemplate: function() {
        return validateTemplate;
    },
    getValidTemplates: function() {
        return getValidTemplates;
    }
});
const _log = require("../utils/log");
function validateTemplate(templateName) {
    const validTemplates = getValidTemplates();
    if (!validTemplates.map((t)=>t.name).includes(templateName)) {
        (0, _log.error)(`'${templateName}' is not a valid template.`);
        (0, _log.info)(`Valid templates: ${validTemplates.map((t)=>t.name).join(', ')}`);
        return false;
    }
    return true;
}
function getValidTemplates() {
    return [
        {
            name: 'blank',
            description: 'Blank Template',
            type: 'starter',
            url: 'https://github.com/payloadcms/payload/templates/blank'
        },
        {
            name: 'website',
            description: 'Website Template',
            type: 'starter',
            url: 'https://github.com/payloadcms/payload/templates/website'
        },
        {
            name: 'ecommerce',
            description: 'E-commerce Template',
            type: 'starter',
            url: 'https://github.com/payloadcms/payload/templates/ecommerce'
        },
        {
            name: 'plugin',
            description: 'Template for creating a Payload plugin',
            type: 'plugin',
            url: 'https://github.com/payloadcms/payload-plugin-template'
        },
        {
            name: 'payload-demo',
            description: 'Payload demo site at https://demo.payloadcms.com',
            type: 'starter',
            url: 'https://github.com/payloadcms/public-demo'
        },
        {
            name: 'payload-website',
            description: 'Payload website CMS at https://payloadcms.com',
            type: 'starter',
            url: 'https://github.com/payloadcms/website-cms'
        }
    ];
}
