// create-payload-app/dist/utils/messages.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    welcomeMessage: function() {
        return welcomeMessage;
    },
    helpMessage: function() {
        return helpMessage;
    },
    successMessage: function() {
        return successMessage;
    }
});
const _chalk = /*#__PURE__*/ _interop_require_default(require("chalk"));
const _figures = /*#__PURE__*/ _interop_require_default(require("figures"));
const _path = /*#__PURE__*/ _interop_require_default(require("path"));
const _terminallink = /*#__PURE__*/ _interop_require_default(require("terminal-link"));
const _templates = require("../lib/templates");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const header = (message)=>`${_chalk.default.yellow(_figures.default.star)} ${_chalk.default.bold(message)}`;
const welcomeMessage = (0, _chalk.default)`
  {green Welcome to Payload. Let's create a project! }
`;
const spacer = ' '.repeat(8);
function helpMessage() {
    const validTemplates = (0, _templates.getValidTemplates)();
    return (0, _chalk.default)`
  {bold USAGE}

      {dim $} {bold npx create-payload-app}
      {dim $} {bold npx create-payload-app} my-project
      {dim $} {bold npx create-payload-app} -n my-project -t blog

  {bold OPTIONS}

      -n     {underline my-payload-app}         Set project name
      -t     {underline template_name}          Choose specific template

        {dim Available templates: ${formatTemplates(validTemplates)}}

      --use-npm                     Use npm to install dependencies
      --use-yarn                    Use yarn to install dependencies
      --use-pnpm                    Use pnpm to install dependencies
      --no-deps                     Do not install any dependencies
      -h                            Show help
`;
}
function formatTemplates(templates) {
    return `\n\n${spacer}${templates.map((t)=>`${t.name}${' '.repeat(28 - t.name.length)}${t.description}`).join(`\n${spacer}`)}`;
}
function successMessage(projectDir, packageManager) {
    return `
  ${header('Launch Application:')}

    - cd ${projectDir}
    - ${packageManager === 'yarn' ? 'yarn' : 'npm run'} dev or follow directions in ${createTerminalLink('README.md', `file://${_path.default.resolve(projectDir, 'README.md')}`)}

  ${header('Documentation:')}

    - ${createTerminalLink('Getting Started', 'https://payloadcms.com/docs/getting-started/what-is-payload')}
    - ${createTerminalLink('Configuration', 'https://payloadcms.com/docs/configuration/overview')}

`;
}
// Create terminalLink with fallback for unsupported terminals
function createTerminalLink(text, url) {
    return (0, _terminallink.default)(text, url, {
        fallback: (text, url)=>`${text}: ${url}`
    });
}
